app/controllers/exams_controller.rb

```diff
@@ -67,4 +88,23 @@ def set_exam
     def exam_params
       params.require(:exam).permit(:name, :max_num_questions, :max_duration)
     end
+    
+    #Creates assembled exam questions
+    def assemble_exam_questions(practice_exam)
+      selected_questions = Question.all.sample(10)
+      selected_questions.each do |question|
+        choices = question.question_choices.to_a
+        correct_choice = choices.find(&:is_correct)
+        choices.delete(correct_choice)
+        choices.shuffle!
+        selected_choices = [correct_choice] + choices.take(3)
+        selected_choices.each do |choice|
+          AssembledExamQuestion.create(
+            practice_exam: practice_exam,
+            question: question,
+            question_choice: choice
+          )
+        end
+      end
+    end
 end
```

Comment:
With a method this long, you can split it into multiple methods.

Reply:
Thanks for the advice, I split it!